src/connection.ts

```diff
@@ -535,6 +536,13 @@ export interface ConnectionOptions {
    */
   connectionRetryInterval?: number;
 
+  /**
+   * Custom connector factory method.
+   *
+   * (default: `undefined`)
+   */
+  connector?: () => Promise<import('net').Socket>;
```

Comment:
I think we import `Socket` from `net` somewhere further up, so you can just change the type to `Promise<Socket>`. What do you think?

Reply:
right! that makes sense! let me change it 😊